File: VB/piesCustomExport/Form1.vb
Imports Microsoft.VisualBasic
Imports DevExpress.DashboardCommon
Imports DevExpress.DashboardWin
Imports DevExpress.XtraCharts
Imports DevExpress.XtraReports.UI
Imports System
Imports System.Collections.Generic
Imports System.ComponentModel
Imports System.Data
Imports System.Drawing
Imports System.Linq
Imports System.Text
Imports System.Threading.Tasks
Imports System.Windows.Forms

Namespace piesCustomExport
	Partial Public Class Form1
		Inherits Form
		Public Sub New()
			InitializeComponent()
			dashboardDesigner1.CreateRibbon()
			dashboardDesigner1.LoadDashboard("nwind.xml")
		End Sub

		Private Sub dashboardDesigner1_DashboardItemControlUpdated(ByVal sender As Object, ByVal e As DevExpress.DashboardWin.DashboardItemControlEventArgs) Handles dashboardDesigner1.DashboardItemControlUpdated
			If e.ChartControl IsNot Nothing Then
				Dim pieItem As PieDashboardItem = TryCast(dashboardDesigner1.Dashboard.Items(e.DashboardItemName), PieDashboardItem)
				If pieItem Is Nothing Then
					Return
				End If
				For Each series As Series In e.ChartControl.Series
					TryCast(series.View, PieSeriesView).TotalLabel.Visible = True
					TryCast(series.View, PieSeriesView).TotalLabel.TextPattern = "Total {TV:c0}"
				Next series
			End If
		End Sub

		Private Sub dashboardDesigner1_CustomExport(ByVal sender As Object, ByVal e As CustomExportEventArgs) Handles dashboardDesigner1.CustomExport
			For Each printControl In e.GetPrintableControls()
				If TypeOf printControl.Value Is XRChart Then
					Dim pieItemName = printControl.Key
					Dim dashboardControl As IDashboardControl = CType(sender, IDashboardControl)
					Dim pieDashboardItem As PieDashboardItem = TryCast(dashboardControl.Dashboard.Items(pieItemName), PieDashboardItem)
					If pieDashboardItem Is Nothing Then
						Return
					End If
					Dim pieChart As XRChart = TryCast(printControl.Value, XRChart)
					For Each series As Series In pieChart.Series
						TryCast(series.View, PieSeriesView).TotalLabel.Visible = True
						TryCast(series.View, PieSeriesView).TotalLabel.TextPattern = "Total {TV:c0}"
					Next series
				End If
			Next printControl
		End Sub
	End Class
End Namespace

File: CS/piesCustomExport/Form1.cs
using DevExpress.DashboardCommon;
using DevExpress.DashboardWin;
using DevExpress.XtraCharts;
using DevExpress.XtraReports.UI;
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using System.Windows.Forms;

namespace piesCustomExport
{
    public partial class Form1 : Form
    {
        public Form1()
        {
            InitializeComponent();
            dashboardDesigner1.CreateRibbon();
            dashboardDesigner1.LoadDashboard(@"..\..\data\nwind.xml");
        }

        private void dashboardDesigner1_DashboardItemControlUpdated(object sender, DevExpress.DashboardWin.DashboardItemControlEventArgs e)
        {
            if (e.ChartControl != null)
            {
                PieDashboardItem pieItem = dashboardDesigner1.Dashboard.Items[e.DashboardItemName] as PieDashboardItem;
                if (pieItem == null) return;
                foreach (Series series in e.ChartControl.Series)
                {
                    (series.View as PieSeriesView).TotalLabel.Visible = true;
                    (series.View as PieSeriesView).TotalLabel.TextPattern = "Total {TV:c0}";
                }
            }
        }

        private void dashboardDesigner1_CustomExport(object sender, CustomExportEventArgs e)
        {
            foreach (var printControl in e.GetPrintableControls())
            {
                if (printControl.Value is XRChart)
                {
                    var pieItemName = printControl.Key;
                    IDashboardControl dashboardControl = (IDashboardControl)sender;
                    PieDashboardItem pieDashboardItem = dashboardControl.Dashboard.Items[pieItemName] as PieDashboardItem;
                    if (pieDashboardItem == null) return;
                    XRChart pieChart = printControl.Value as XRChart;
                    foreach (Series series in pieChart.Series)
                    {
                        (series.View as PieSeriesView).TotalLabel.Visible = true;
                        (series.View as PieSeriesView).TotalLabel.TextPattern = "Total {TV:c0}";
                    }
                }
            }
        }
    }
}
